apps/core/src/hooks/affine/use-language-helper.ts

```diff
@@ -0,0 +1,27 @@
+import { LOCALES, useI18N } from '@affine/i18n';
+import { useCallback, useMemo } from 'react';
+
+export function useLanguageHelper() {
+  const i18n = useI18N();
+  const currentLanguage = useMemo(
+    () => LOCALES.find(item => item.tag === i18n.language),
+    [i18n.language]
+  );
+  const languagesList = useMemo(
+    () =>
+      LOCALES.map(item => ({
+        tag: item.tag,
+        originalName: item.originalName,
+        name: item.name,
+      })),
+    []
+  );
+  const onSelect = useCallback(
+    (event: string) => {
+      i18n.changeLanguage(event);
+    },
+    [i18n]
+  );
+
+  return { languagesList, currentLanguage, onSelect };
```

Comment:
need to wrap this with `useMemo` otherwise it will cause commands to register/unregister randomly

Reply:
Thank you for the guidance, I'll check and amend it tomorrow.